Travis error looks unrelated:

```
RuntimeError: module compiled against API version 0xb but this version of numpy is 0xa
Traceback (most recent call last):
  File "/home/travis/miniconda3/envs/pandas/lib/python3.6/site-packages/_pytest/config.py", line 342, in _getconftestmodules
    return self._path2confmods[path]
KeyError: local('/home/travis/build/pandas-dev/pandas/pandas')
```

--- a/pandas/tests/util/test_util.py
+++ b/pandas/tests/util/test_util.py
@@ -471,7 +471,7 @@ class TestLocaleUtils(object):
 
 def test_make_signature():
     # See GH 17608
-    make_signature(validate_kwargs)
     # Case where the func does not have default kwargs
-    make_signature(deprecate_kwarg)
+    make_signature(validate_kwargs)
     # Case where the func does have default kwargs
+    make_signature(deprecate_kwarg)

--- a/pandas/tests/util/test_util.py
+++ b/pandas/tests/util/test_util.py
@@ -472,6 +472,12 @@ class TestLocaleUtils(object):
 def test_make_signature():
     # See GH 17608
     # Case where the func does not have default kwargs
-    make_signature(validate_kwargs)
+    sig = make_signature(validate_kwargs)
+    assert sig == (['fname', 'kwargs', 'compat_args'],
+                   ['fname', 'kwargs', 'compat_args'])
+
     # Case where the func does have default kwargs
-    make_signature(deprecate_kwarg)
+    sig = make_signature(deprecate_kwarg)
+    assert sig == (['old_arg_name', 'new_arg_name',
+                    'mapping=None', 'stacklevel=2'],
+                   ['old_arg_name', 'new_arg_name', 'mapping', 'stacklevel'])

